v0/destinations/google_adwords_remarketing_lists/transform.js

```diff
@@ -0,0 +1,272 @@
+const {
+  isDefinedAndNotNullAndNotEmpty,
+  returnArrayOfSubarrays,
+  CustomError,
+  constructPayload,
+  defaultRequestConfig,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+const {
+  offlineDataJobsMapping,
+  addressInfoMapping,
+  BASE_ENDPOINT,
+  attributeMapping,
+  hashAttributes
+} = require("./config");
+
+const hashEncrypt = object => {
+  Object.keys(object).forEach(key => {
+    if (hashAttributes.includes(key)) {
+      // eslint-disable-next-line no-param-reassign
+      object[`${key}`] = sha256(object[`${key}`]);
+    }
+  });
+};
+
+/**
+ * This function is used for building the response. It create a default rudder response
+ * and populate headers, params and body.JSON
+ * @param {*} metadata
+ * @param {*} body
+ * @param {*} param2
+ * @returns
+ */
+const responseBuilder = (metadata, body, { Config }) => {
+  const payload = body;
+  const response = defaultRequestConfig();
+  response.endpoint = `${BASE_ENDPOINT}/${Config.customerId}/offlineUserDataJobs`;
+  response.body.JSON = removeUndefinedAndNullAndEmptyValues(payload);
+  const accessToken = metadata.secret.access_token;
+  response.params = { listId: Config.listId, customerId: Config.customerId };
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json",
+    "developer-token": get(metadata, "secret.developer_token")
+  };
+  if (Config.subAccount)
+    if (Config.loginCustomerId)
+      response.headers["login-customer-id"] = Config.loginCustomerId;
+    else
+      throw new CustomError(
+        `[Google_adwords_remarketing_list]:: loginCustomerId is required as subAccount is true.`,
+        400
+      );
+  return response;
+};
+/**
+ * This function helps creates an array with proper mapping for userIdentiFier.
+ * Logics: Here we are creating an array with all the attributes provided in the add/remove array
+ * inside listData.
+ * @param {rudder event message properties listData add} attributeArray
+ * @param {rudder event destination} Config
+ * @returns
+ */
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const userIdentifier = [];
+  const { typeOfList } = Config;
+  const { isHashRequired } = Config;
+  let attribute;
+  // setting attribute name according to type of list
+  switch (typeOfList) {
+    case "userID":
+      attribute = "thirdPartyUserId";
+      break;
+    case "mobileDeviceID":
+      attribute = "mobileId";
+      break;
+    default:
+      attribute = Config.userSchema;
+  }
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach((element, index) => {
+      if (isHashRequired) {
+        hashEncrypt(element);
+      }
+      // checking if the attribute is an array or not for generic type list
+      if (!Array.isArray(attribute)) {
+        if (element[`${attribute}`]) {
```

Comment:
same question here. looks like `atrribute` is of type string. I want to understand why we are using string templating here